File: homework/hw3/c++/hex_mcts_test.cpp
#include "algorithm/monte_carlo_tree_search.hpp"
#include "problem/hex.hpp"

int main(){

    HexState<11> t;

    while (not t.done()){

        MonteCarloTreeSearch<decltype(t)> mcts(t);

        auto action = mcts.select_action(5000, 0.2);
        std::cout << action << std::endl;
        t = t.next(action);

        t.show();
    }

    return 0;
}

File: homework/hw3/python/hex_mcts_test.py
from algorithm.monte_carlo_tree_search import MonteCarloTreeSearch
from problem.hex import HexState

if __name__ == '__main__':
    t = HexState(11)
    while not t.done():
        mcts = MonteCarloTreeSearch(t)
        action = mcts.select_action(150, 0.2)
        print(action)
        t = t.next(action)
        t.show()
